### PR TITLE
Xml writer

### PR DESCRIPTION
**Pull request by mennoo1996**
Extended XML writer to write camera timelines and cameras.

### CODE REVIEW

File: src/main/java/data/CameraTimeline.java [hunk 1]
```diff
@@ -12,6 +14,9 @@
     // The camera that is associated with this timeline.
     @Getter @Setter
     private Camera camera;
+    
+    @Getter
+    private ArrayList<CameraShot> shots;
```
Comment (bart-de-jonge) [inline src/main/java/data/CameraTimeline.java:19]: The TimeLine class already contains a list of shots, can you use that? If not you should remove that and also add a list of DirectorShots to the DIrectorTimeline

File: src/test/java/data/CameraShotTest.java [hunk 2]
```diff
@@ -32,5 +32,17 @@ public void setNameTest() {
...
+    @Test
+    public void getDescriptionTest() {
+        assertEquals("some-description", cameraShot.getDescription());
+        
+    }
+    
+    @Test
+    public void setDescriptionTest() {
+        cameraShot.setDescription("a");
+        assertEquals("a", cameraShot.getDescription());
```
Comment (geenen124) [inline src/test/java/data/CameraShotTest.java:45]: Are tests for lombok generated getters/setters included in the coverage?

File: src/main/java/xml/XmlWriter.java [hunk 3]
```diff
@@ -20,9 +23,14 @@
...
+
 import org.w3c.dom.Document;
 import org.w3c.dom.Element;
 
+
+
+
+
 /**
  * Created by Bart.
```
Comment (Markvanderuit) [inline src/main/java/xml/XmlWriter.java:35]: Maybe add javadoc with class description?

File: src/test/java/xml/XmlWriterTest.java [hunk 4]
```diff
@@ -33,10 +33,11 @@ public void writeProject() throws Exception {
         project.setDirectorTimeline(dt);
```
Comment (bart-de-jonge) [inline src/test/java/xml/XmlWriterTest.java:63]: Shouldn't you add some cameraTimelines and cameras here to really test the added functionalities?

File: src/main/java/data/CameraTimeline.java (hunk 1 above)
Comment (mennoo1996) [inline src/main/java/data/CameraTimeline.java:19]: The DirectorTimeline already has a list of DirectorShots, which is why I did the same for CameraTimeline. I could either remove both of these and use the list from TimeLine, or leave it as it is now

File: src/test/java/data/CameraShotTest.java (hunk 2 above)
Comment (mennoo1996) [inline src/test/java/data/CameraShotTest.java:45]: I believe @martijn9612 tested this and found that this is indeed the case

File: src/main/java/xml/XmlWriter.java (hunk 3 above)
Comment (mennoo1996) [inline src/main/java/xml/XmlWriter.java:35]: Will do

**Comment by bart-de-jonge** [comment]
Look great, please process the code review comments and then we can merge this.

File: src/test/java/xml/XmlWriterTest.java (hunk 4 above)
Comment (mennoo1996) [inline src/test/java/xml/XmlWriterTest.java:63]: Yes, I'll do that

File: src/main/java/data/CameraTimeline.java (hunk 1 above)
Comment (bart-de-jonge) [inline src/main/java/data/CameraTimeline.java:19]: Okay please delete the list of shots from the TimeLine class then, that is not used anymore

**Comment by bart-de-jonge** [comment]
Nice work, if the checks pass we can merge

**Comment by geenen124** [comment]
Looks good!

**Pull request merged**